pkg/build/api/types.go

```diff
@@ -352,6 +356,25 @@ type SourceBuildStrategy struct {
 	ForcePull bool
 }
 
+// BuildHookSpec holds build hooks with actions grouped by build status.
+type BuildHookSpec struct {
+	// The OnSuccess group is a list of BuildHooks that will be executed
+	// when the build changes to BuildPhaseComplete.
+	OnSuccess []BuildHook
+
+	// The OnFailure group is a list of BuildHooks that will be executed
+	// when the build changes to BuildPhaseFailed.
+	OnFailure []BuildHook
+}
+
+// A BuildHook defines actions to be executed when a build changes to a certain
+// phase.
+type BuildHook struct {
+	// StartBuilds holds a list of references to BuildConfigs that will be
+	// started when the hook is executed.
+	StartBuilds []kapi.ObjectReference
+}
```

Comment:
> now we can support any phase the user wants, not just success/fail

I don't think we should support arbitrary hooks like that as of now, unless we have clear use cases to execute an action when, e.g., the build transitions to `BuildPhaseRunning`.
And we'd need to rework the transitions in code, because they happen in multiple places, unlike the transition to `BuildPhaseComplete` and `BuildPhaseFailed` that happens only in `HandlePod` (now, at least).